src/CSnakes.Tests/GeneratedSignatureTests.cs

```diff
@@ -226,4 +252,43 @@ public void TestUnionReturnType(string returnType)
         var module1 = ModuleReflection.MethodsFromFunctionDefinitions(functions1).ToImmutableArray();
         Assert.Single(module1);
     }
+
+    [Theory]
+    [InlineData("""
+        
+                def foo(  # csharp: ignore
+                        bar,
+                        baz,
+                    ):
+                    pass
+        
+        """,
+        Label = "C# ignore comment - Multi lined function definition")]
+    [InlineData("""
+        
+                def hello(): # csharp: ignore ...
+        
+        """,
+        Label = "C# ignore comment - Single line function definition")]
+    [InlineData("""
+        
+                def foo():  # type: ignore # csharp: ignore
+                    pass
+        
+        """,
+        Label = "C# ignore comment - Embedded within a comment after")]
+    [InlineData("""
+        
+                def bar():  # csharp: ignore # type: ignore
+                    pass
+        
+        """,
+        Label = "C# ignore comment - Embedded within a comment before")]
+    public void TestCSharpIgnoreCommentFunction(string code)
+    {
+        SourceText sourceText = SourceText.From(code);
+        Assert.True(PythonParser.TryParseFunctionDefinitions(sourceText, out var functions, out var errors));
+        Assert.Empty(errors);
+        Assert.Empty(functions);
+    }
```

Comment:
Considering asserting that the input had no common indentation (perhaps with a comment as to why):

```suggestion
        var indentations =
            from line in sourceText.Lines
            select line.ToString()
            into line
            where line.Length > 0
            select line.TakeWhile(ch => ch == ' ').Count();
        Assert.Equal(0, indentations.Min());
    }
```